packages/vite/src/node/plugins/esbuild.ts

```diff
@@ -170,7 +172,7 @@ export async function transformWithEsbuild(
 
 export function esbuildPlugin(options: ESBuildOptions = {}): Plugin {
   const filter = createFilter(
-    options.include || /\.(tsx?|jsx)$/,
+    options.include || /\.([cm]?ts|[jt]sx)$/,
```

Comment:
I think we could remove `.cts` handling here too as it's not valid in source code. We should probably only support `.mts` in source code, and `.cts`/`.mts` for deps.

```suggestion
    options.include || /\.(m?ts|[jt]sx)$/,
```